contracts/zero-ex/contracts/src/transformers/WethTransformer.sol

```diff
@@ -0,0 +1,98 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "../errors/LibTransformERC20RichErrors.sol";
+import "./Transformer.sol";
+import "./LibERC20Transformer.sol";
+
+
+/// @dev A transformer that wraps or unwraps WETH.
+contract WethTransformer is
+    Transformer
+{
+    /// @dev Transform data to ABI-encode and pass into `transform()`.
+    struct TransformData {
+        // The token to wrap/unwrap. Must be either ETH or WETH.
+        IERC20TokenV06 token;
+        // Amount of `token` to wrap or unwrap.
+        // `uint(-1)` will unwrap the entire balance.
```

Comment:
I think we should try to make these API's as explicit as possible. For example, we could standardize that if a transformer has more than one action we'll use an Enum to differentiate:
```
enum Action {
    UNWRAP,
    WRAP
}
```

Reply:
idk, is `1` or `0` any more explicit than passing in 2 distinct addresses?

Reply:
Are you thinking of like a separate transformer for wrapping and unwrapping? That's probably fine too.

Reply:
Btw I think the confusion just arises here because we're overloading the `uint(-1)`. I think this is generally used to mean like the whole balance, but here we also use it to flip the operation. I don't feel too strongly either way, but worth considering.

Reply:
We don't use `uint256(-1)` to flip the operation, we specify either the WETH address or the ETH address (0xeee...).

Reply:
`{token: 0xeeee..., amount: uint256(-1)`: Wrap all ETH.
`{token: WETH_ADDRESS, amount: uint256(-1)`: Unwrap all WETH.

Reply:
Ah okay, that makes more sense. +1